### PR TITLE
Feature/wakayama 1276 add places endpoints

### PR DESCRIPTION
**Pull request by oleg-n-k**
- Get countries and cities endpoints
- Reverse geocoding improvements

### CODE REVIEW

File: HappyTravel.Wakayama.Api/Services/ReverseGeocodingService.cs [hunk 1]
```diff
@@ -18,22 +18,22 @@ public ReverseGeocodingService(ElasticGeoServiceClient geoServiceClient, IOption
     }
     
 
-    public async Task<ReverseGeocodingResponse> Search(ReverseGeocodingRequest request, CancellationToken cancellationToken)
+    public async Task<ResponseBuilder> Search(ReverseGeocodingRequest request, CancellationToken cancellationToken)
     {
-        const int maxSearchDistanceInKm = 5;
+        const int maxSearchDistanceInKm = 10;
```
Comment (hasmikharutyunyan) [inline HappyTravel.Wakayama.Api/Services/ReverseGeocodingService.cs:23]: maybe have this value in constant ?

File: HappyTravel.Wakayama.Api/Services/IPlaceSearchingService.cs [hunk 2]
```diff
@@ -0,0 +1,9 @@
+using HappyTravel.Wakayama.Api.Models;
+
+namespace HappyTravel.Wakayama.Api.Services;
+
+public interface IPlaceSearchingService
```
Comment (hasmikharutyunyan) [inline HappyTravel.Wakayama.Api/Services/IPlaceSearchingService.cs:5]: maybe just `IPlaceService` ?

File: HappyTravel.Wakayama.Api/Infrastructure/Extensions/MultiLanguageExtensions.cs [hunk 3]
```diff
@@ -0,0 +1,29 @@
+using HappyTravel.LocationNameNormalizer.Extensions;
+using HappyTravel.Wakayama.Common.Models;
+
+namespace HappyTravel.Wakayama.Api.Infrastructure.Extensions;
+
+public static class MultiLanguageExtensions
+{
+    public static string? GetValue(this MultiLanguage? value)
```
Comment (hasmikharutyunyan) [inline HappyTravel.Wakayama.Api/Infrastructure/Extensions/MultiLanguageExtensions.cs:8]: I think better use not nullable MultiLanguage

File: HappyTravel.Wakayama.Api/Models/ResponseBuilder.cs [hunk 4]
```diff
@@ -0,0 +1,6 @@
+namespace HappyTravel.Wakayama.Api.Models;
+
+public class ResponseBuilder
```
[the remark this answers is not in the document]
Comment (oleg-n-k) [reply HappyTravel.Wakayama.Api/Models/ResponseBuilder.cs]: fixed

File: HappyTravel.Wakayama.Api/Infrastructure/Extensions/MultiLanguageExtensions.cs (hunk 3 above)
Comment (oleg-n-k) [reply HappyTravel.Wakayama.Api/Infrastructure/Extensions/MultiLanguageExtensions.cs:8]: Wakayama has own model for multi language properties. Some of them might be null

File: HappyTravel.Wakayama.Api/Services/ReverseGeocodingService.cs (hunk 1 above)
Comment (oleg-n-k) [reply HappyTravel.Wakayama.Api/Services/ReverseGeocodingService.cs]: This section has been refactored. Please check this again

File: HappyTravel.Wakayama.Api/Services/ReverseGeocodingService.cs [hunk 5]
```diff
@@ -10,57 +10,37 @@ namespace HappyTravel.Wakayama.Api.Services;
...
-            }
-            coordinateToSearchIndexes = emptyResponseIndexes;
-            
-            if (!coordinateToSearchIndexes.Any())
-                break;
-        }
 
-        return _reverseGeocodingResponseBuilder.Build(searchResponseStore);
+    public async Task<ReverseGeoCodingResponse> Get(ReverseGeocodingRequest request, CancellationToken cancellationToken)
+    {
```
Comment (hasmikharutyunyan) [inline HappyTravel.Wakayama.Api/Services/ReverseGeocodingService.cs:33]: Are we need validation for coordinates here?

File: HappyTravel.Wakayama.Api/Services/ReverseGeocodingService.cs [hunk 6]
```diff
@@ -129,8 +149,50 @@ private SearchRequest<Place> CreateSearchRequest(GeoPoint point, Distance distan
...
+            
+            if (!notFoundIndexes.Any())
+                break;
+        }
+
+        return searchResponseStore;
+    }
+
+
+    private List<int> GetNotFoundIndexesAndFillStore(MultiSearchResponse searchResponse, ref Dictionary<string, Place> responseStore, List<int> notFoundIndexes)
```
Comment (hasmikharutyunyan) [inline HappyTravel.Wakayama.Api/Services/ReverseGeocodingService.cs:178]: maybe  `GetNotFoundIndexesOrFillStore` ?

Comment (oleg-n-k) [reply HappyTravel.Wakayama.Api/Services/ReverseGeocodingService.cs:179]: rename to GetNotFoundIndexesAndAddResponseDataToStore

File: HappyTravel.Wakayama.Api/Services/ReverseGeocodingService.cs (hunk 5 above)
Comment (oleg-n-k) [reply HappyTravel.Wakayama.Api/Services/ReverseGeocodingService.cs:33]: It ignores wrong coordinates so as not to stop the batch processing

File: HappyTravel.Wakayama.Api/Infrastructure/Extensions/MultiLanguageExtensions.cs (hunk 3 above)
Comment (oleg-n-k) [reply HappyTravel.Wakayama.Api/Infrastructure/Extensions/MultiLanguageExtensions.cs:8]: replaced with a helper method

**Review by hasmikharutyunyan — APPROVED** [review]

**Review by AlexeyShok — APPROVED** [review]

**Pull request merged**